src/background/isolation.ts

```diff
@@ -588,4 +592,48 @@ export class Isolation {
     }
     return false;
   }
+
+  // Moved logic from commands.ts to create this utility function.
+  // preferences.ts might be able to use this as well, but I'm not sure how the Vue persistence might get in the way
```

Comment:
Thanks for moving this from commands, makes it cleaner!

Did you encounter issues while trying to reuse this from `preferences.ts` as well? I think it should be fine to reuse - would just move out the `storage.persist()` (back into `commands.ts`), so that persisting doesn't happen twice in `preferences.ts`.

Reply:
Though, just realized that it might not make that much sense to call `setIsolation` from `handleChanges` since the preference was already set by the UI. Maybe splitting the badge/interval logic into another method inside `isolation.ts`, like `handleChange` and reuse that?

Reply:
I wasn't confident enough in the persistence flow to test the change in preferences.ts.  Since both commands.ts and now isolation.ts require the explicit persist, and preferences.ts doesn't, how do feel about a persistence flag on the method call?  Or, maybe there's no harm in a double-persist.

I'll think about your great observation on the interrupted auto-re-isolate (by closing the browser), and also your suggestion here and run some tests.

I'm also adding a few test cases to try to improve code coverage in isolation.  Mocha/Chai really doesn't seem to like delaying tests to allow the auto-re-isolate to trigger.  I've tried a dozen combinations of what should work, and can't seem to find the right combo.  Grr.

Reply:
> how do feel about a persistence flag on the method call? Or, maybe there's no harm in a double-persist.

I'd like to avoid not needed persisting when possible. The idea was to add two methods to the `Isolation` class, something like `toggleActive` and `handleActiveToggled`.

`toggleActive` could be called from commands, which then switches the active preference, calls `handleActiveToggled` and finally persists the storage. Then from the `handlesChanges` method inside `Preferences` it could just call `handleActiveToggled`. Does that make sense?

> I'll think about your great observation on the interrupted auto-re-isolate (by closing the browser), and also your suggestion here and run some tests.

Thanks! If you have questions about initialization lifecycle, feel free to ask.

> I'm also adding a few test cases to try to improve code coverage in isolation. Mocha/Chai really doesn't seem to like delaying tests to allow the auto-re-isolate to trigger. I've tried a dozen combinations of what should work, and can't seem to find the right combo. Grr.

Tests are not really needed in this case - but if you want to add some, that'd be even better!

The [container cleanup (removal) tests use timers](https://github.com/stoically/temporary-containers/blob/cb180f2bb9e5d58e44707437d6f02a8ea62cf5af/test/background.cleanup.test.ts#L8), which is [sinon fake timers](https://sinonjs.org/releases/latest/fake-timers/) under the hood. That might be what you're looking for in this case.

Reply:
> `toggleActive` could be called from commands, which then switches the active preference, calls `handleActiveToggled` and finally persists the storage. Then from the `handleChanges` method inside `Preferences` it could just call `handleActiveToggled` (which handles badge/interval). Does that make sense?

This technique is working nicely.  Commit coming soon, after I address the initial firefox shutdown interruption idea, and look a bit deeper into #433